File: api/src/main/java/api/security/service/TokenExtractor.java
package api.security.service;

import api.security.config.JwtSecurityConfig;
import api.security.exceptions.JwtAuthenticationException;

import javax.servlet.http.HttpServletRequest;

public class TokenExtractor {

    private static final String PREFIX = "Bearer ";

    public static String extract(HttpServletRequest request){

        String header = request.getHeader(JwtSecurityConfig.TOKEN_HEADER);

        if (header == null || !header.startsWith(PREFIX)) {
            throw new JwtAuthenticationException("Token is missing");
        }

        return header.substring(7);
    }
}

File: api/src/main/java/api/security/service/TokenExtractor.kt
package api.security.service

import api.security.config.JwtSecurityConfig
import api.security.exceptions.JwtAuthenticationException

import javax.servlet.http.HttpServletRequest

private const val PREFIX = "Bearer "

fun extract(request: HttpServletRequest): String {

    val header = request.getHeader(JwtSecurityConfig.TOKEN_HEADER)

    if (header == null || !header.startsWith(PREFIX)) {
        throw JwtAuthenticationException("Token is missing")
    }

    return header.substring(PREFIX.length + 1)
}
